### DISCUSSION TITLE
select all/remove all option in MultiSelect

### DISCUSSION THREAD

**Discussion by dakahn** [carbon-design-system/carbon#7622] [Ideas]
(original issue https://github.com/carbon-design-system/carbon/issues/4593)

### Summary

There is an option to remove all the selected items from the multi select list, however if a user selected all the items and then clicked on remove all by mistake (for instance, if there is a narrow dropdown showing a list of 2 digit codes or something similar its not hard to press the remove all button by mistake), there is no way to select all with one click other than by clicking on individual items. In our case, we have drop downs with about a thousand items. 

### Justification

Business reason is that if there is an option to remove all selections, there should be an option to select all to undo if you remove all selections by mistake.

### Desired UX and success metrics

A sample can be seen on the [mdbootstrap multiselect](https://mdbootstrap.com/docs/jquery/forms/multiselect/) page.

### "Must have" functionality

There should be a way to select all the items in multiselect list with one click

**Comment by jnm2377** [comment]
I think this is definitely a fair ask. We see this in DataTable (select all on the toolbar). We also have an expand all option for expandable data table. I don't know what the design for this would look like. @carbon-design-system/design do y'all have any thoughts?

**Comment by aagonzales** [comment]
Yeah I think that seems fair to me as well. I wonder if it can be an optional feature or if it should always be there?

**Comment by laurenmrice** [comment]
I could see it as being an optional prop you can turn on. Deepending on if there are like 5 items vs. 25+ to choose from, people can decide to use a select all if they want to. We have gotten a lot of asks for this in the past.

**Comment by jnm2377** [comment]
Optional seems like a good idea, since some people might not need or want it.
